src/BasicAccessFactory.php

```diff
@@ -37,10 +39,19 @@ public function __invoke(ContainerInterface $container) : BasicAccess
             );
         }
 
+        /** @var callable|null $responseFactory */
+        $responseFactory = $container->get(ResponseInterface::class) ?? null;
```

Comment:
You might want to require webmozart/assert and use Assert::isInstanceOf() instead of inline annotation 👍

Reply:
It will be BC change, isn't it?

`webmozart/assert` will throw `\InvalidArgumentException`
In current code, we throw own package exception `Mezzio\Authentication\Exception\InvalidConfigException`

Reply:
I think the assertion is not needed here: there's an explicit `if ()` block after this, so the `/** @var` block can also be removed in this case.